tools/distpartitioning/utils.py

```diff
@@ -557,3 +557,21 @@ def map_partid_rank(partid, world_size):
         id.
     """
     return partid % world_size
+
+
+def generate_read_list(num_files, world_size):
+    """Generate the file IDs to read for each rank.
```

Comment:
Can you give an example in the docstring, as this API will be used by external users.

Reply:
Below is added.
```
Examples
    --------
    >>> tools.distpartitionning.utils.generate_read_list(10, 4)
    [array([0, 1, 2]), array([3, 4, 5]), array([6, 7]), array([8, 9])]
```